documentation/solution-parser.md

```diff
@@ -0,0 +1,91 @@
+# Solution Parser
+
+## Expander
+
+`Expander` class is used to expand value from an expression, which can
+be function invocation. These expression (or embedded properties)
+have form - "$(property)".
+
+`Expander` handles intrinsic functions, which is a list of built-in functions
+(bunch of ifs).
+
+MSBuild defines initial set of properties, like `MSBuildFrameworkToolsPath32`,
+`MSBuildFrameworkToolsPath`, etc.
```

Comment:
```suggestion
MSBuild defines an initial set of properties like `MSBuildFrameworkToolsPath32`, `MSBuildFrameworkToolsPath`, etc.
```

"a built-in set" rather than "an initial set"? Unsure which is better.